From 7056e8ecf1802754d57fbfe9f3fabe4f59d0805e Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Tue, 13 Sep 2022 17:39:34 +0530
Subject: [PATCH] ath12k: Fixed dma unmapping in tx desc cleanup

In ath12k_dp_cc_cleanup() in tx desc cleanup part
dma_unmap_single(ab->dev, ATH12K_SKB_RXCB(skb)->paddr,
			skb->len + skb_tailroom(skb), DMA_FROM_DEVICE);
is used which is incorrect.

It should be
dma_unmap_single(ab->dev, ATH12K_SKB_CB(skb)->paddr,
                          skb->len, DMA_TO_DEVICE);

because in ath12k_dp_tx() (skb_cb = ATH12K_SKB_CB(skb)) we are mapping it as below
ti.paddr = dma_map_single(ab->dev, skb->data, skb->len, DMA_TO_DEVICE);
skb_cb->paddr = ti.paddr;

fix: 9015303005f50 ("ath12k: allocate tx descriptors per pool")

Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1194,7 +1194,7 @@ static void ath12k_dp_cc_cleanup(struct
 			if (!skb)
 				continue;
 
-			dma_unmap_single(ab->dev, ATH12K_SKB_RXCB(skb)->paddr,
+			dma_unmap_single(ab->dev, ATH12K_SKB_CB(skb)->paddr,
 					 skb->len, DMA_TO_DEVICE);
 			dev_kfree_skb_any(skb);
 		}
